spinach/brokers/redis.py

```diff
@@ -272,19 +289,33 @@ def enqueue_jobs_from_dead_broker(self, dead_broker_id: uuid.UUID) -> int:
             self._to_namespaced(ALL_BROKERS_HASH_KEY),
             self._to_namespaced(ALL_BROKERS_ZSET_KEY),
             self.namespace,
-            self._to_namespaced(NOTIFICATIONS_KEY)
+            self._to_namespaced(NOTIFICATIONS_KEY),
+            self._to_namespaced(MAX_CONCURRENCY_KEY),
+            self._to_namespaced(CURRENT_CONCURRENCY_KEY),
         )
 
     def register_periodic_tasks(self, tasks: Iterable[Task]):
         """Register tasks that need to be scheduled periodically."""
-        tasks = [task.serialize() for task in tasks]
+        _tasks = [task.serialize() for task in tasks]
         self._number_periodic_tasks = len(tasks)
         self._run_script(
             self._register_periodic_tasks,
             math.ceil(datetime.now(timezone.utc).timestamp()),
             self._to_namespaced(PERIODIC_TASKS_HASH_KEY),
             self._to_namespaced(PERIODIC_TASKS_QUEUE_KEY),
-            *tasks
+            *_tasks
+        )
+
+    def set_concurrency_keys(self, tasks: Iterable[Task]):
+        """For each retryable Task, set up its concurrency keys."""
+        _tasks = [task.serialize() for task in tasks if task.max_retries > 0]
```

Comment:
Could you walk me through the reasoning behind filtering by `task.max_retries > 0`?

Reply:
It's a thinko, I meant to use `max_concurrency`. I've fixed it and added a test.